src/form-builder.tsx

```diff
@@ -1,23 +1,72 @@
 import { DisplayProvider } from "./context-providers";
 import { BuilderProvider } from "./context-providers/builder";
-import { Container, ToastWrapper, Wrapper } from "./form-builder.styles";
+import {
+    Container,
+    ErrorWrapper,
+    ToastWrapper,
+    Wrapper,
+} from "./form-builder.styles";
+import { ErrorDisplay } from "@lifesg/react-design-system/error-display";
 
 export const FormBuilder = () => {
-    return (
-        <BuilderProvider>
-            <DisplayProvider>
-                <Wrapper>
-                    <Container type="grid" stretch>
-                        <ToastWrapper>
-                            <Toasts />
-                        </ToastWrapper>
-                        <MainPanel />
-                        <SidePanel />
-                    </Container>
-                </Wrapper>
-            </DisplayProvider>
-        </BuilderProvider>
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+    const [isLargeScreen, setIsLargeScreen] = useState(
+        window.innerWidth >= 1200
     );
+
+    // =========================================================================
+    // EFFECTS
+    // =========================================================================
+    useEffect(() => {
+        const handleResize = () => {
+            setIsLargeScreen(window.innerWidth >= 1200);
+        };
+
+        window.addEventListener("resize", handleResize);
+        return () => {
+            window.removeEventListener("resize", handleResize);
+        };
+    }, []);
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+    const renderScreen = () => {
+        if (isLargeScreen) {
+            return (
+                <BuilderProvider>
+                    <DisplayProvider>
+                        <Wrapper>
+                            <Container type="grid" stretch>
+                                <ToastWrapper>
+                                    <Toasts />
+                                </ToastWrapper>
+                                <MainPanel />
+                                <SidePanel />
+                            </Container>
+                        </Wrapper>
+                    </DisplayProvider>
+                </BuilderProvider>
+            );
+        } else {
+            return (
+                <ErrorWrapper>
+                    <ErrorDisplay
+                        type="warning"
+                        title={"Screen size not supported"}
+                        description={
+                            "Expand to a bigger screen size or switch to desktop view to use the form builder."
+                        }
+                    />
+                </ErrorWrapper>
+            );
```

Comment:
can we move this to a new component so we can keep the logic in this file simple?

also I dont think we need a separate `renderScreen()` function since this is the only render logic here